contracts/signature-drop/SignatureDrop.sol

```diff
@@ -325,21 +297,18 @@ contract SignatureDrop is
         return hasRole(DEFAULT_ADMIN_ROLE, _msgSender());
     }
 
+    /// @dev Returns whether platform fee info can be set in the given execution context.
+    function _canSetClaimConditions() internal view override returns (bool) {
+        return hasRole(DEFAULT_ADMIN_ROLE, _msgSender());
+    }
+
     /*///////////////////////////////////////////////////////////////
                         Miscellaneous
     //////////////////////////////////////////////////////////////*/
 
     /// @dev Burns `tokenId`. See {ERC721-_burn}.
     function burn(uint256 tokenId) public virtual {
-        address ownerOfToken = ownerOf(tokenId);
-        //solhint-disable-next-line max-line-length
-        require(
-            _msgSender() == ownerOfToken ||
-                isApprovedForAll(ownerOfToken, _msgSender()) ||
-                getApproved(tokenId) == _msgSender(),
-            "caller not owner nor approved"
-        );
-        _burn(tokenId);
+        _burn(tokenId, true);
```

Comment:
make a comment here that `_burn` internally check for approvals from erc721a

Reply:
Will add consideration in a code comment + design document.